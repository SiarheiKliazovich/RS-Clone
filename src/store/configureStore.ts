/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
