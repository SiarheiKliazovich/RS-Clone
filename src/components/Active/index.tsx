import { cloneElement, FunctionComponent } from "react";
import { useSelector } from "react-redux";
import { ActiveType } from "../../types";
import { IStore } from "../../interfaces/store";

const Active: FunctionComponent<ActiveType> = ({ classN, children }) => {
  const { accentColor } = useSelector((state: IStore) => state.appInterface);

  return (
    <>
      {classN.indexOf("users__header-button_active") !== -1
        ? cloneElement(children, {
            style: {
              backgroundColor: accentColor.hover,
            },
          })
        : cloneElement(children, {
            style: {
              backgroundColor: accentColor.static,
            },
          })}
    </>
  );
};

export default Active;
