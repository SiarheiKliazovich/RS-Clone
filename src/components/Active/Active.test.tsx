import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Active from ".";

const mockStore = configureStore([]);

describe("Active", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Active classN="users__header-button">
          <button>Test Button</button>
        </Active>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
