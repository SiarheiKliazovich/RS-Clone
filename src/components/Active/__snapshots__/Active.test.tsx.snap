// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active renders correctly 1`] = `
<button
  style="background-color: rgb(0, 0, 0);"
>
  Test Button
</button>
`;
